### DISCUSSION TITLE
Secure adblocker recommendation for Safari browser

### DISCUSSION THREAD

**Discussion by hgj44d** [PrivSec-dev/privsec.dev#154] [Q&A] [answered]
On the site [Badness Enumeration](https://privsec.dev/posts/knowledge/badness-enumeration/) there is a part:
> If you insist on getting an adblocker however, I highly recommend that you use purely declarative, permission less Manifest V3 ones like [uBlock Origin Lite](https://chrome.google.com/webstore/detail/ublock-origin-lite/ddkjiahejlhfcafbddmgiahcphecmpfh). While they do block fewer ads and trackers than their Manifest V2 counterparts and V3 extensions with “Read and change all your data on all websites”, they pose much less of a threat to your privacy and security while still providing the convenience of blocking annoyances.

Is there anything similar for the Safari browser on Apple devices (MacOS and iOS)? What about [Adguard](https://adguard.com/en/welcome.html) (seems to be the most popular adblocker for Safari)?

My goal is **not** to gain more security or privacy through an adblocker, I only want to get rid of annoying advertisements and get some quality-of-life improvments.

**Comment by beerisgood** [comment, marked as answer]
> I only want to get rid of annoying advertisements and get some quality-of-life improvments.

Did you try the Reader mode?
https://support.apple.com/guide/safari/hide-ads-when-reading-sfri32632/mac

**Reply by hgj44d** [reply]
Thanks! That's a great feature, but:
* it only works on suported sites and not on all sites
* ~~it's not enabled all the time and I have to manually enable it on every site~~

**Reply by beerisgood** [reply]
In website settings you can set reader mode as default or add sites manually.

**Comment by wj25czxj47bu6q** [comment]
The free version of AdGuard for iOS uses the Safari content blocker API and thus should be rather safe to use.

I have no idea about the situation with Safari on MacOS, but you do have the option of installing any Chromium browser or Firefox to use uBlock Origin Lite or Brave Browser which has a built-in ad blocker.

Reader view seems like it could be a good solution if it works well.

**Comment by Ganwtrs** [comment]
There is now uBlock Origin Lite for iOS, and it can be more secure than AdGuard especially when using it in Basic mode
